JS-1-week-4/voice.js

```diff
@@ -0,0 +1,121 @@
+let userName = "";
+const toDo = [];
+const monthNames = [
+  "January",
+  "February",
+  "March",
+  "April",
+  "May",
+  "June",
+  "July",
+  "August",
+  "September",
+  "October",
+  "November",
+  "December",
+];
+const numericalRegex = /\d/;
+
+function getReply(command) {
+  // Handling name introduction
+  if (command.toLowerCase().startsWith("hello my name is ")) {
+    userName = command.slice(17).trim();
+    if (userName && userName.toLowerCase() === name.toLowerCase()) {
+      return `You've already introduced yourself, ${userName}.`;
+    } else {
+      return `Nice to meet you, ${userName}.`;
+    }
+  }
+  if (command.toLowerCase().startsWith("what is my name")) {
+    return `your name is ${userName}`;
+  } else if (userName == "") {
+    return `your name is not defined`;
+  } else if (
+    command.toLowerCase().startsWith("add") &&
+    command.toLowerCase().match("to my todo")
+  ) {
+    let todoStr = command.slice(4, -11);
+    toDo.push(todoStr);
+    return `${todoStr} added to your todo`;
+  } else if (
+    command.toLowerCase().startsWith("remove") &&
+    command.toLowerCase().match("from my todo")
+  ) {
+    let todoStr = command.slice(7, -13);
+    toDo.pop(todoStr);
+    return `Removed ${todoStr} from your todo`;
+  } else if (command.toLowerCase() === "what is on my todo?") {
+    let todoSize = toDo.length;
+    let todoStr = todoSize > 1 ? toDo.join(" and ") : toDo[0];
+    return `you have ${todoSize} todos - ${todoStr}`;
+  } else if (command.toLowerCase() === "what day is it today?") {
+    let date = new Date();
+    const day = date.getDate();
+    const month = monthNames[date.getMonth()];
+    const year = date.getFullYear();
+
+    return `${day}. of ${month} ${year}`;
+  } else if (
+    numericalRegex.test(command) &&
+    command.toLowerCase().startsWith("what is")
+  ) {
+    const additionRegex = /what is (\d+)\s*\+\s*(\d+)/i;
```

Comment:
Using a regex is a brilliant idea! If you are up for the challenge, you can implement this solution with a single regex 😉

Reply:
Hi @vdaguenet , sorry for late reply, can you please help me with this!!

Reply:
This was purely optional. You can carry on and merge this PR without changing this section of the homework.

The idea was to update the regex by adding all the operators to a group as you did for the numbers. Then use it like this:
```js
const regex = /.../;
const matches = command.match(additionRegex);
const num1 = parseInt(matches[1], 10);
const operator = matches[2];
const num2 = parseInt(matches[3], 10);

switch(operator) {
  case '+':
    return num1 + num2;
  case '-':
    return num1 - num2;
  // and so on...
}
``` 

I recommend using this website: [regex101](https://regex101.com/) to try out your regex over different values.

Reply:
@vdaguenet , thanks for your  advice , I will practice on it.